flink-ml-lib/src/main/java/org/apache/flink/ml/common/param/HasL2.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.ml.common.param;
+
+import org.apache.flink.ml.param.DoubleParam;
+import org.apache.flink.ml.param.Param;
+import org.apache.flink.ml.param.ParamValidators;
+import org.apache.flink.ml.param.WithParams;
+
+/** Interface for the shared L2 param. */
+public interface HasL2<T> extends WithParams<T> {
+    Param<Double> L_2 = new DoubleParam("l2", "The l2 param.", 0.1, ParamValidators.gt(0.0));
```

Comment:
L1 and L2 are the hyper parameters for ftrl. It has different meaning from regParam.

Reply:
@weibozhao Currently the class `LogisticRegression` pass the value of `getReg()` to `LogisticGradient::l2`, which suggests that getReg() actually represents the hyper-parameter l2 for `LogisticRegression`.

Could you explain why we can not get the value of `l2` from `getReg())` for `OnlineLogisticRegression`?

Reply:
According to the Elastic net regularization wiki page [1], l1 and l2 could be theoretically translated to a regularization constant and the SVM regularization constant (i.e. the elastic net). Thus the regularization constant is different from l1 and l2 under this context

As far as the mathematical formula is concerned, is the existing `HasReg::REG` the same as L2 or the regularization constant explained in [1]? If it is the same as L2, we probably should rename this class to e.g. `HasL2`. Otherwise, we probably should rename the variable `LogisticGradient::l2` as appropriate.

[1] https://en.wikipedia.org/wiki/Elastic_net_regularization#:~:text=In%20statistics%20and%2C%20in%20particular,the%20lasso%20and%20ridge%20methods.

Reply:
After some offline discussion, @weibozhao  and I agree that we could reuse the exising `HasReg` and `HasElasticNet` param, without introducing `HasL1` and `HasL2`.

Reply:
done